lib/client.js

```diff
@@ -322,7 +322,9 @@ exports.createClient = function(reference, name) {
             if (reportStats) {
               reportStats(true, 'http-' + res.status);
             }
-            if(res.headers['content-type'].indexOf('application/json') === -1 || !res.body){
+            if((_.includes(res.headers, 'content-type') &&
```

Comment:
`_.includes(res.headers, 'content-type')` checks if there is a value called 'content-type', not if there is a key.

Either use `_.includes(res.headers['content-type'], 'application/json')` or do `if (res.headers['content-type'] && res.headers['content-type'].indexOf(...`

That's just my guess...